File: src/targets/php/http2/fixtures/application-form-encoded.php
<?php

$client = new http\Client;
$request = new http\Client\Request;

$body = new http\Message\Body;
$body->append(new http\QueryString([
  'foo' => 'bar',
  'hello' => 'world'
]));

$request->setRequestUrl('https://httpbin.org/anything');
$request->setRequestMethod('POST');
$request->setBody($body);

$request->setHeaders([
  'content-type' => 'application/x-www-form-urlencoded'
]);

$client->enqueue($request)->send();
$response = $client->getResponse();

echo $response->getBody();

File: src/targets/rust/reqwest/fixtures/application-form-encoded.rs
use serde_json::json;
use reqwest;

#[tokio::main]
pub async fn main() {
    let url = "https://httpbin.org/anything";

    let payload = json!({
        "foo": "bar",
        "hello": "world"
    });

    let client = reqwest::Client::new();
    let response = client.post(url)
        .header("content-type", "application/x-www-form-urlencoded")
        .form(&payload)
        .send()
        .await;

    let results = response.unwrap()
        .json::<serde_json::Value>()
        .await
        .unwrap();

    println!("{}", results);
}
